tests/components/top_bar.test.ts

```diff
@@ -309,6 +309,20 @@ describe("TopBar component", () => {
       await click(fixture, `.o-menu-item-button[title="${iconTitle}"]`);
       expect(model.getters.getCurrentStyle()).toEqual(expectedStyle);
     });
+    test.each([["Left (Ctrl+Shift+L)"], ["Center (Ctrl+Shift+E)"], ["Right (Ctrl+Shift+R)"]])(
+      "Update icon on change",
+      async (iconTitle) => {
```

Comment:
You should directly test against the actions and not look at how they are rendered. I suggest you look at how Lucas did it here : https://github.com/odoo/o-spreadsheet/pull/2651/files#diff-e729fc703acf88b1fee2adb7d8a03b2081192d64b7ae73e52dc7cee34fc7d306R879  He's testing the dynamic change of the key `description` so it's really close to what you are doing :)

Reply:
Hello @rrahir ,I attempted the approach you recommended. However, due to the absence of a registry for action buttons on the toolbar, I couldn't utilize the getNode() as it uses topbarmenuregistry function as shown in Lucas's pull request. Nevertheless, I devised an alternative method to associate the icon with the current cell style. Please take a look.